components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/par/OAuth2ParEndpoint.java

```diff
@@ -265,12 +268,45 @@ private void validateRequestObject(OAuthAuthzRequest oAuthAuthzRequest) throws P
                 if (requestObject == null) {
                     throw new ParClientException(OAuth2ErrorCodes.INVALID_REQUEST, ParConstants.INVALID_REQUEST_OBJECT);
                 }
+                if (OAuth2Util.isFapiConformantApp(oAuthAuthzRequest.getClientId())) {
+                    validatePKCEParameters(requestObject, paramMap);
+                }
             }
-        } catch (RequestObjectException e) {
+        } catch (RequestObjectException | IdentityOAuth2Exception e) {
             if (OAuth2ErrorCodes.SERVER_ERROR.equals(e.getErrorCode())) {
                 throw new ParCoreException(e.getErrorCode(), e.getMessage(), e);
             }
             throw new ParClientException(e.getErrorCode(), e.getMessage(), e);
         }
     }
+
+    private void validatePKCEParameters(RequestObject requestObject, Map<String, List<String>> paramMap)
+            throws ParClientException {
+
+        String codeChallenge = null;
+        String codeChallengeMethod = null;
+        if (paramMap != null) {
+            if (paramMap.containsKey(OAuthConstants.OAUTH_PKCE_CODE_CHALLENGE)) {
+                codeChallenge = paramMap.get(OAuthConstants.OAUTH_PKCE_CODE_CHALLENGE).get(0);
```

Comment:
Add a method to override all required parameters

Reply:
fixed